docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -1507,6 +1507,51 @@ As the core components use emotion as a styled engine, the props used by emotion
   }
   ```
 
+# createGenerateClassName
+
+- The `createGenerateClassName` is no longer exported from `@material-ui/core/styles`. You should import it directly from `@material-ui/styles`.
+
+  ```diff
+  -import { createGenerateClassName } from '@material-ui/core/styles';
+  +import { createGenerateClassName } from '@material-ui/styles';
+  ```
+
+# jssPreset
+
+- The `jssPreset` is no longer exported from `@material-ui/core/styles`. You should import it directly from `@material-ui/styles`.
+
+  ```diff
+  -import { jssPreset } from '@material-ui/core/styles';
+  +import { jssPreset } from '@material-ui/styles';
+  ```
+
+# ServerStyleSheets
+
+- The `ServerStyleSheets` is no longer exported from `@material-ui/core/styles`. You should import it directly from `@material-ui/styles`.
+
+  ```diff
+  -import { ServerStyleSheets } from '@material-ui/core/styles';
+  +import { ServerStyleSheets } from '@material-ui/styles';
+  ```
+
+# StylesProvider
+
+- The `StylesProvider` is no longer exported from `@material-ui/core/styles`. You should import it directly from `@material-ui/styles`.
+
+  ```diff
+  -import { StylesProvider } from '@material-ui/core/styles';
+  +import { StylesProvider } from '@material-ui/styles';
+  ```
+
+# useThemeVariants
```

Comment:
Has this ever been public?

Reply:
I saw that it was exported. We can even remove it, with the new styles engine is not required anymore.

Reply:
Would leave it in the end when we cleanup the `@material-ui/styled` package, I'd just remove it from `@material-ui/core/styles` right now.